ui/DesignSystem/Component/Dropdown.svelte

```diff
@@ -1,22 +1,38 @@
-<script>
+<script lang="ts">
   import { Icon } from "../Primitive";
+  import type { EmojiAvatar } from "../../src/avatar";
+
   import Option from "./Dropdown/Option.svelte";
 
-  export let placeholder = null;
+  type Option =
+    | { value: string; title: string }
+    | {
+        value: string;
+        title: string;
+        avatarProps: {
+          avatarFallback: EmojiAvatar;
+          title: string;
+          variant?: "circle" | "square";
+        };
+      };
+
+  export let options: Option[];
+
+  export let placeholder = "";
 
-  export let dataCy = null;
-  export let options = null;
-  export let style = null;
-  export let optionStyle = null;
-  export let valid = true;
-  export let validationMessage = null;
-  export let validationPending = false;
+  export let dataCy = "";
+  export let style = "";
+  export let optionStyle = "";
+
+  export let valid: boolean = true;
```

Comment:
Could we convert these booleans into variants? For example,
```
enum Input {
  Raw(string),
  Valid(string),
}
```

Although really we'd like a newtype `Valid` that has a **smart constructor** `string -> Result<Valid, ValidationError>`

Reply:
Interesting topic - we actually have typed states for validation: https://github.com/radicle-dev/radicle-upstream/blob/master/ui/src/validation.ts#L11

I'll make an issue for adding this to `Dropdown.svelte` because this pattern already exists for the other inputs.

Reply:
Sounds good to me :+1: